crates/rspack_util/src/swc.rs

```diff
@@ -26,3 +34,51 @@ fn test_normalize_custom_filename() {
   let expected_output = "custom_filename";
   assert_eq!(normalize_custom_filename(input), expected_output);
 }
+
+/**
+ * Some code is modified based on
+ * https://github.com/swc-project/swc/blob/e6fc5327b1a309eae840fe1ec3a2367adab37430/crates/swc_compiler_base/src/lib.rs#L342
+ * Apache-2.0 licensed
+ * Author Donny/강동윤
+ * Copyright (c)
+ */
+pub fn minify_file_comments(
+  comments: &SingleThreadedComments,
+  preserve_comments: BoolOr<JsMinifyCommentOption>,
+  preserve_annotations: bool,
+) {
+  match preserve_comments {
+    BoolOr::Bool(true) | BoolOr::Data(JsMinifyCommentOption::PreserveAllComments) => {}
+
+    BoolOr::Data(JsMinifyCommentOption::PreserveSomeComments) => {
+      let preserve_excl = |_: &BytePos, vc: &mut std::vec::Vec<Comment>| -> bool {
+        // Preserve license comments.
+        //
+        // See https://github.com/terser/terser/blob/798135e04baddd94fea403cfaab4ba8b22b1b524/lib/output.js#L175-L181
+        vc.retain(|c: &Comment| {
+          c.text.contains("@lic")
+            || c.text.contains("@preserve")
+            || c.text.contains("@copyright")
+            || c.text.contains("@cc_on")
+            || (preserve_annotations
```

Comment:
I may not understand this.

Isn't `preserve_annotations` alway set to `false` with `preserve_comment` set to `JsMinifyCommentOption::PreserveSomeComments`?

Reply:
Yes, this is aligned with current implementation and `swc-loader` behavior. `preserve_annotations` is enabled only when `comments` is set to `"all"`.